src/xenia/kernel/xam/user_profile.cc

```diff
@@ -10,25 +10,53 @@
 #include "xenia/kernel/xam/user_profile.h"
 
 #include <sstream>
+#include <xhash>
 
 #include "third_party/fmt/include/fmt/format.h"
 #include "xenia/kernel/kernel_state.h"
 #include "xenia/kernel/util/shim_utils.h"
 
+DEFINE_string(gamertag_0, "User_0", "The name of your user profile.",
+              "Profile");
+DEFINE_string(gamertag_1, "User_1", "The name of your user profile.",
+              "Profile");
+DEFINE_string(gamertag_2, "User_2", "The name of your user profile.",
+              "Profile");
+DEFINE_string(gamertag_3, "User_3", "The name of your user profile.",
+              "Profile");
+
 namespace xe {
 namespace kernel {
 namespace xam {
 
 UserProfile::UserProfile(uint8_t index) {
+  const std::vector<std::string> username_list = {
+      cvars::gamertag_0,
+      cvars::gamertag_1,
+      cvars::gamertag_2,
+      cvars::gamertag_3,
+  };
+
+  std::string current_username = username_list[index];
+  if (current_username.empty()) {
+    current_username = "User_" + std::to_string(index);
+  }
+
+  // TODO(Gliniak): Username sanitizer
+  //
   // 58410A1F checks the user XUID against a mask of 0x00C0000000000000 (3<<54),
   // if non-zero, it prevents the user from playing the game.
   // "You do not have permissions to perform this operation."
-  xuid_ = 0xB13EBABEBABEBABE + index;
-  name_ = "User";
-  if (index) {
-    name_ = "User_" + std::to_string(index);
-  }
 
+  // Get assigned username to current index from config
+  // Then use lower 4 bytes based on player name for unique xuid
+  size_t hash = std::hash<std::string>{}(current_username);
```

Comment:
I think this makes sense. Am I correct in saying you cannnot change the offline profile name on an Xbox?

Reply:
Sadly it is possible to change gamertag locally, however I'm not sure how to deal with storing XUID and Gamertag. 

For stfs packages that would be easy because they're stored in package header, but we don't have such possibility now.

Reply:
Hmm, ok. I don't know too much about the offline side of things.

But yeah for the online side of things, you have an STFS package with the "Account" file contianing the online gamertag and XUID. I did the same for the online fork, so people could copy paste their account file from a real console and use their existing gamertag and XUID. For new profiles, the XUID is randomised (with some high-ish bits to avoid conflicts with real Xbox Live XUIDs), but stored to file for future runs. But all this was without the full STFS package - just the one file.

I guess if we wanted to avoid a full STFS package, we could just generate a header file or something (content\XUID\FFFE07D1\Headers\00010000\...`), and support name changes in-app (menu option or similar) rather than config settings.